src/tv2-common/videoSwitchers/TriCaster.ts

```diff
@@ -175,15 +189,96 @@ export class TriCaster extends VideoSwitcherBase {
 		)
 	}
 
-	public getDveTimelineObjects(_properties: DveProps): TSR.TSRTimelineObj[] {
-		throw new Error('Method not implemented.')
-	}
-
 	public updateUnpopulatedDveBoxes(
-		_timelineObject: TSR.TSRTimelineObj,
-		_input: number | SpecialInput
+		timelineObject: TSR.TSRTimelineObj,
+		input: number | SpecialInput
 	): TSR.TSRTimelineObj {
-		throw new Error('Method not implemented.')
+		if (!this.isDveBoxes(timelineObject)) {
+			this.logWrongTimelineObjectType(timelineObject, this.updateUnpopulatedDveBoxes.name)
+			return timelineObject
+		}
+
+		const layers: Partial<Record<TSR.TriCasterLayerName, TSR.TriCasterLayer>> = (
+			(timelineObject as TSR.TimelineObjTriCasterME).content.me as TSR.TriCasterMixEffectInEffectMode
+		).layers!
```

Comment:
This part is messy. There are a lot of casting in between accessors and nullable values.

The typecast of `timelineObject` can be avoided, by changing the return type of `isDveBoxes`to `timelineObject is TSR.TimelineObjTriCasterME`.
By splitting the access chain of `layers` up into two steps, we can get type castings and access operations that are easier to understand.
I'll suggest doing something like this:

```typescript
public isDveBoxes = (timelineObject: TimelineObjectCoreExt<unknown, unknown>): timelineObject is TSR.TimelineObjTricasterMe => {
    // ...
}

public updateUnpopulatedDveBoxes(
    timelineObject: TSR.TSRTimelineObj,
    input: number | SpecialInput
): TSR.TSRTimelineObj {
    if (!this.isDveBoxes(timelineObject)) {
        // handle incorrect timeline object type
        return timelineObject
    }
  
    const dveMixEffect = timelineObject.content.me as TSR.TriCasterMixEffectInEffectMode
    const layers = dveMixEffect.layers as NonNullable<TSR.TricasterMixEffectInEffectMode['layers']>
    
    // ...
}
```